src/flamegraph/color/mod.rs

```diff
@@ -16,49 +16,98 @@ const BLUE_GRADIENT: (&str, &str) = ("#eeeeee", "#e0e0ff");
 const GREEN_GRADIENT: (&str, &str) = ("#eef2ee", "#e0ffe0");
 const GRAY_GRADIENT: (&str, &str) = ("#f8f8f8", "#e8e8e8");
 
+/// A flame graph background color.
+///
+/// The default background color usually depends on the color scheme:
+///
+///  - [`BasicPalette::Mem`] defaults to [`BackgroundColor::Green`].
+///  - [`BasicPalette::Io`] and [`MultiPalette::Wakeup`] default to [`BackgroundColor::Blue`].
+///  - [`BasicPalette::Hot`] defaults to [`BackgroundColor::Yellow`].
+///  - All other [`BasicPalette`] variants default to [`BackgroundColor::Grey`].
+///  - All other [`MultiPalette`] variants default to [`BackgroundColor::Yellow`].
+///
+/// `BackgroundColor::default()` is `Yellow`.
 #[derive(Clone, Copy, Debug)]
 pub enum BackgroundColor {
+    /// A yellow gradient from `#EEEEEE` to `#EEEEB0`.
     Yellow,
+    /// A blue gradient from `#EEEEEE` to `#E0E0FF`.
     Blue,
+    /// A green gradient from `#EEF2EE` to `#E0FFE0`.
     Green,
+    /// A grey gradient from `#F8F8F8` to `#E8E8E8`.
     Grey,
+    /// A flag background color with the given RGB components.
+    ///
+    /// Expressed in string form as `#RRGGBB` where each component is written in hexadecimal.
     Flat(u8, u8, u8),
 }
 
+impl Default for BackgroundColor {
+    fn default() -> Self {
+        BackgroundColor::Yellow
+    }
+}
+
+/// A flame graph color palette.
+///
+/// Defaults to [`BasicPalette::Hot`].
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum Palette {
+    /// A plain color palette in which the color is independent of the function name.
     Basic(BasicPalette),
+    /// A semantic color palette in which different hues are used to signifiy semantic aspects of
+    /// different function names (kernel functions, JIT functions, etc.).
     Multi(MultiPalette),
 }
 
+impl Default for Palette {
+    fn default() -> Self {
+        Palette::Basic(BasicPalette::Hot)
+    }
+}
+
+/// A plain color palette in which the color is independent of the function name.
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum BasicPalette {
+    /// A palette in which colors are chosen randomly from a red-yellow spectrum.
```

Comment:
`randomly` <- here again the random aspect is not bound to the palette (only the options `--hash` and/or `--cp`)